extensions/flight-sql/src/main/java/io/deephaven/server/flightsql/FlightSqlResolver.java

```diff
@@ -651,22 +654,39 @@ private Table executeSqlQuery(SessionState session, String sql) {
         final Map<String, Table> queryScopeTables =
                 (Map<String, Table>) (Map) queryScope.toMap(queryScope::unwrapObject, (n, t) -> t instanceof Table);
         final TableSpec tableSpec = Sql.parseSql(sql, queryScopeTables, TicketTable::fromQueryScopeField, null);
-        // Note: we only technically need the lock if any of the source tables are refreshing, but that requires some
-        // deeper introspection. We will play it safe and take the shared lock regardless.
-        final Lock sharedLock = executionContext.getUpdateGraph().sharedLock();
-        sharedLock.lock();
+
+        // We could consider doing finer-grained sharedLock in the future; right now, taking it for the whole operation
+        // if any of the TicketTable sources are refreshing.
+        boolean hasRefreshingSource = false;
+        for (final TableSpec node : ParentsVisitor.reachable(List.of(tableSpec))) {
+            // Of the source tables, SQL can produce a NewTable or a TicketTable (until we introduce custom functions,
+            // where we could conceivable have it produce EmptyTable, TimeTable, etc).
+            if (!(node instanceof TicketTable)) {
+                continue;
+            }
+            final String variableName = new String(((TicketTable) node).ticket(), StandardCharsets.UTF_8).substring(2);
+            final Table sourceTable = queryScopeTables.get(variableName);
```

Comment:
This doesn't seem like the right way to resolve a ticket, generally. I remember that SQL is limited to named tables, and so I guess FlightSQL is limited to scope tickets?

Reply:
Yes; but you are correct, it is better to use the resolver here.